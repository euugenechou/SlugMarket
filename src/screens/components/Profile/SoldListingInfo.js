// React imports
import React, { Component } from "react";
import { View, Text, StyleSheet, Image } from "react-native";
import { Button } from "react-native-elements";

import { API, Auth } from "aws-amplify"

export default class SoldListingsInfo extends Component {
  static navigationOptions = () => ({
    headerTintColor: "white",
    headerStyle: {
      height: 40,
      backgroundColor: "white",
      shadowColor: "transparent",
      borderBottomWidth: 0
    },
    headerBackground: (
      <Image
        style = { StyleSheet.absoluteFill }
        source = { require("../../../assets/textbooks.jpg") }
      />
    ),
  });

  state = {
    userId: "",
    timeAdded: this.props.navigation.getParam("timeAdded"),
    category: this.props.navigation.getParam("category"),
    itemName: this.props.navigation.getParam("itemName"),
    price: this.props.navigation.getParam("price"),
    seller: this.props.navigation.getParam("seller"),
    description: this.props.navigation.getParam("description"),
    isSold: this.props.navigation.getParam("isSold"),
    isRemoved: this.props.navigation.getParam("isRemoved"),
    phoneNumber: this.props.navigation.getParam("phoneNumber"),
    email: this.props.navigation.getParam("email")
  };

  getUpdatedPostObject() {
    let postObject = { body: this.state };
    return postObject;
  }

  async deletePost() {
    const userInfo = await Auth.currentUserInfo().catch(error => {
      Alert.alert(JSON.stringify(error));
      return;
    });
    const apiName = "itemPostingsCRUD";
    const path = "/itemPostings";
    const updatedObject = this.getUpdatedPostObject();
    updatedObject.body.isRemoved = true;
    API.put(apiName, path, updatedObject)
      .then(res => {
        console.log(res);
        this.props.navigation.navigate("MainProfile");
      })
      .catch(err => {
        Alert.alert("Error updating post. Please try again");
        console.log(err);
      });
  }

  render() {
    return (
      <View style = { styles.container }>
        <Text style = { styles.titleText }>
          { this.props.navigation.getParam("itemName") }
        </Text>
        <Text style = { styles.text }>
          ${ this.props.navigation.getParam("price") }
        </Text>
        <Text style = { styles.text }>
          { this.props.navigation.getParam("category") }
        </Text>
        <Text style = { styles.text }>
          { this.props.navigation.getParam("description") }
        </Text>
        <Button
            raised
            color = "white"
            backgroundColor = "red"
            borderRadius = { 5 }
            fontWeight = "bold"
            onPress = { () => this.deletePost() }
            title = "Delete Item"
            containerViewStyle = { styles.buttonContainer }
          />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  text: {
    fontSize: 22,
    color: "black",
    fontWeight: "400",
    alignSelf: 'center',
    paddingBottom: 10,
    paddingHorizontal: 30,
  },
  titleText: {
    fontSize: 30,
    color: "black",
    fontWeight: "800",
    alignSelf: 'center',
    paddingBottom: 10,
    marginTop: 100
  },
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white"
  },
  buttonContainer: {
    width: 300,
    alignSelf: "center",
    paddingVertical: 20
  }
});
