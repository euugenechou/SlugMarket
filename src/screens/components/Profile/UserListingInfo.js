/* React imports */
import React, { Component } from "react";
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  Alert,
  ScrollView
} from "react-native";

/* AWS imports */
import { API, Auth } from "aws-amplify";

import { Button } from "react-native-elements";

class UserListingsInfo extends Component {
  state = {
    itemName: this.props.navigation.getParam("itemName"),
    price: this.props.navigation.getParam("price"),
    seller: this.props.navigation.getParam("seller"),
    category: this.props.navigation.getParam("category"),
    description: this.props.navigation.getParam("description"),
    timeAdded: this.props.navigation.getParam("timeAdded")
  };

  static navigationOptions = () => ({
    headerTintColor: "teal",
    headerStyle: {
      height: 40,
      backgroundColor: "white",
      shadowColor: "transparent",
      borderBottomWidth: 0,
    }
  });

  async updateItemFields() {
    const userInfo = await Auth.currentUserInfo().catch(error => {
      Alert.alert(JSON.stringify(error));
      return;
    });
    const apiName = "itemPostingsCRUD";
    const path = "/itemPostings";
    const updatedObject = this.getUpdatedPostObject(userInfo.id);
    API.put(apiName, path, updatedObject)
      .then(res => {
        console.log(res);
        this.props.navigation.navigate("MainProfile");
      })
      .catch(err => {
        Alert.alert("Error updating post. Please try again");
        console.log(err);
      });
  }

  getUpdatedPostObject(cognitoUserId) {
    let postObject = { body: this.state };
    postObject.body.userId = cognitoUserId;
    return postObject;
  }

  async deletePost() {
    const userInfo = await Auth.currentUserInfo().catch(error => {
      Alert.alert(JSON.stringify(error));
      return;
    });
    const apiName = "itemPostingsCRUD";
    const path = "/itemPostings";
    const updatedObject = this.getUpdatedPostObject(userInfo.id);
    updatedObject.body.isSold = true;
    API.put(apiName, path, updatedObject)
      .then(res => {
        console.log(res);
        this.props.navigation.navigate("MainProfile");
      })
      .catch(err => {
        Alert.alert("Error updating post. Please try again");
        console.log(err);
      });
  }

  onChangeText(key, value) {
    this.setState({
      [key]: value
    });
  }

  render() {
    return (
      <View style={{ backgroundColor: "white", flex: 1 }}>
        <ScrollView
          contentContainerStyle={{
            backgroundColor: "white",
            paddingTop: 30,
            paddingLeft: 35,
            flex: 1
          }}
        >
          <View>
            <Text style={styles.text}>Item Name</Text>
            <TextInput
              style={styles.input}
              autoCapitalize="none"
              autoCorrect={false}
              defaultValue={this.props.navigation.getParam("itemName")}
              onChangeText={value => this.onChangeText("itemName", value)}
            />
            <Text style={styles.text}>Item Price</Text>
            <TextInput
              style={styles.input}
              autoCapitalize="none"
              autoCorrect={false}
              defaultValue={this.props.navigation.getParam("price")}
              onChangeText={value => this.onChangeText("price", value)}
            />
            <Text style={styles.text}>Item Description</Text>
            <TextInput
              style={styles.input}
              autoCapitalize="none"
              autoCorrect={false}
              defaultValue={this.props.navigation.getParam("description")}
              onChangeText={value => this.onChangeText("description", value)}
            />
          </View>
          <View
            style={{
              backgroundColor: "white",
              paddingTop: 40,
              alignContent: "space-around",
              justifyContent: "space-around"
            }}
          >
            <Button
              raised
              color="white"
              backgroundColor="teal"
              borderRadius={5}
              fontWeight="bold"
              onPress={() => this.updateItemFields()}
              title="Save Changes"
              containerViewStyle={{
                width: 300,
                paddingBottom: 15,
                paddingRight: 25
              }}
            />
            <Button
              raised
              color="white"
              backgroundColor="goldenrod"
              borderRadius={5}
              fontWeight="bold"
              onPress={() => this.deletePost()}
              title="Mark Item As Sold"
              containerViewStyle={{
                width: 300,
                paddingBottom: 15,
                paddingRight: 25
              }}
            />
            <Button
              raised
              color="white"
              backgroundColor="red"
              borderRadius={5}
              fontWeight="bold"
              onPress={() => this.deletePost()}
              title="Delete Item"
              containerViewStyle={{
                width: 300,
                paddingBottom: 15,
                paddingRight: 25
              }}
            />
          </View>
        </ScrollView>
      </View>
    );
  }
}

export default UserListingsInfo;

const styles = StyleSheet.create({
  text: {
    fontSize: 20,
    color: "black",
    fontWeight: "700"
  },
  input: {
    height: 50,
    color: "black",
    borderBottomWidth: 0.8,
    borderBottomColor: "#dddddd",
    marginBottom: 20,
    width: 300
  },
  container: {
    marginTop: 30,
    marginLeft: 30
  }
});
