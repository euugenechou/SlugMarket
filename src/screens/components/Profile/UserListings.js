/* React imports */
import React, { Component } from "react";
import { View, Text, StyleSheet, Image } from "react-native";

class UserListings extends Component {
  render() {
    return (
      <View
        style={{
          flexDirection: "row",
          width: this.props.width - 40,
          height: this.props.width / 2 - 80,
          borderWidth: 0.5,
          borderColor: "#dddddd",
          marginLeft: 20,
          marginTop: 20
        }}
      >
        <View style={{ flex: 2 }}>
          <Image
            style={{
              flex: 2,
              width: null,
              height: null,
              resizeMode: "cover"
            }}
            source={require("../../../assets/textbooks.jpg")}
          />
        </View>
        <View
          style={{
            flex: 1,
            alignItems: "center",
            justifyContent: "space-evenly"
          }}
        >
          <Text
            style={{ alignItems: "center", fontSize: 18, fontWeight: "bold" }}
          >
            {this.props.name}
          </Text>
          <Text style={{ alignItems: "center", fontSize: 18 }}>
            ${this.props.price}
          </Text>
        </View>
      </View>
    );
  }
}

export default UserListings;

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center"
  }
});
