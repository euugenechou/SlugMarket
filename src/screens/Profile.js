// React imports
import React, { Component } from "react";
import {
  StyleSheet,
  SafeAreaView,
  Image,
  Text,
  View,
  Dimensions,
  ScrollView,
  TouchableHighlight,
  RefreshControl
} from "react-native";
import { Button } from "react-native-elements";
// AWS imports
import { API, Auth } from "aws-amplify";
// Local imports
import UserListings from "./components/Profile/UserListings";

const { height, width } = Dimensions.get("window");

export default class Profile extends Component {
  static navigationOptions = ({ navigation }) => ({
    header: null
  });

  constructor(props) {
    super(props);
    this.state = {
      activeIndex: 0,
      postsToRender: [],
      soldPostsToRender: [],
      userName: "",
      userAttributes: {},
      refreshing: false
    };
  }

  static navigationOtions = {
    tabBarIcon: ({ tintColor }) => (
      <Icon name="person" style={{ color: tintColor }} />
    )
  };

  componentWillMount() {
    this.startHeaderHeight = 60;
    this.getUserPosts();
    this.getUserSoldPosts();
    this.getUserInfo();
  }

  // For getting user posts
  getUserPosts() {
    const path = "/itemPostings/userPosts";
    const apiName = "itemPostingsCRUD";
    const headers = {
      headers: {},
      response: true
    };
    API.get(apiName, path, headers)
      .then(response => {
        response.data.sort((a, b) => {
          return new Date(b.timeAdded) - new Date(a.timeAdded);
        });
        this.setState({
          postsToRender: response.data.filter(post => {
            return !post.isSold && !post.isRemoved;
          })
        });
      })
      .catch(() => Alert.alert("error getting user posts"));
  }

  // For getting user sold posts
  getUserSoldPosts() {
    const path = "/itemPostings/userPosts";
    const apiName = "itemPostingsCRUD";
    const headers = {
      headers: {},
      response: true
    };
    API.get(apiName, path, headers)
      .then(response => {
        response.data.sort((a, b) => {
          return new Date(b.timeAdded) - new Date(a.timeAdded);
        });
        this.setState({
          soldPostsToRender: response.data.filter(post => {
            return post.isSold && !post.isRemoved;
          })
        });
      })
      .catch(() => Alert.alert("error getting user sold posts"));
  }

  segmentClicked = index => {
    this.setState({
      activeIndex: index
    });
  };

  getUserInfo() {
    Auth.currentUserInfo()
      .then(res => {
        this.setState({ userAttributes: res.attributes });
        this.setState({ userName: res.attributes.name });
      })
      .catch(() => Alert.alert("error getting user info"));
  }

  _onRefresh = () => {
    this.setState({ refreshing: true });
    this.getUserPosts();
    this.getUserSoldPosts();
    this.setState({ refreshing: false });
  };

  renderUserListings = () => {
    return this.state.postsToRender.map(post => {
      return (
        <TouchableHighlight
          onPress={() =>
            this.props.navigation.navigate("UserListingInfoScreen", {
              itemName: post.itemName,
              price: post.price,
              seller: post.seller,
              category: post.category,
              description: post.description,
              timeAdded: post.timeAdded,
              isSold: post.isSold,
              isRemoved: post.isRemoved
            })
          }
          underlayColor="white"
          key={post.timeAdded}
        >
          <UserListings
            width={width}
            name={post.itemName}
            price={post.price}
            category={post.category}
            seller={post.seller}
          />
        </TouchableHighlight>
      );
    });
  };

  renderUserSoldListings = () => {
    return this.state.soldPostsToRender.map(post => {
      return (
        <TouchableHighlight
          onPress={() =>
            this.props.navigation.navigate("SoldListingInfoScreen", {
              itemName: post.itemName,
              price: post.price,
              seller: post.seller,
              category: post.category,
              description: post.description,
              timeAdded: post.timeAdded,
              isSold: post.isSold,
              isRemoved: post.isRemoved
            })
          }
          underlayColor="white"
          key={post.timeAdded}
        >
          <UserListings
            width={width}
            name={post.itemName}
            price={post.price}
            category={post.category}
            seller={post.seller}
          />
        </TouchableHighlight>
      );
    });
  };

  render() {
    return (
      <SafeAreaView style={{ backgroundColor: "white", flex: 1 }}>
        <ScrollView
          style={{ marginTop: 20 }}
          scrollEventThrottle={16}
          contentContainerStyle={{ backgroundColor: "white" }}
          refreshControl={
            <RefreshControl
              refreshing={this.state.refreshing}
              onRefresh={this._onRefresh}
            />
          }
          showsVerticalScrollIndicator={false}
        >
          <View style={{ backgroundColor: "white" }}>
            <View style={styles.outerContainer}>
              <Image
                source={require("../assets/Slug.png")}
                style={styles.image}
              />
              <View style={styles.columnContainer}>
                <View style={styles.innerContainer}>
                  <Text style={styles.titleText}>{this.state.userName} </Text>
                  <Text style={styles.text}>
                    {this.state.userAttributes.email}
                  </Text>
                  <Text style={styles.text}>
                    {this.state.userAttributes.phone_number}
                  </Text>
                  <Button
                    color="white"
                    backgroundColor="teal"
                    borderRadius={5}
                    title="Sell Item"
                    fontWeight="bold"
                    onPress={() => this.props.navigation.navigate("AddItem")}
                    containerViewStyle={{ width: 150, paddingTop: 10 }}
                  />
                </View>
              </View>
            </View>
            <Text style={styles.listingText}>Your Listings</Text>
            {this.renderUserListings()}
            <Text style={styles.listingText}>Your Sold Listings</Text>
            {this.renderUserSoldListings()}
          </View>
        </ScrollView>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  outerContainer: {
    backgroundColor: "white",
    flexDirection: "row-reverse",
    flex: 1
  },
  columnContainer: {
    backgroundColor: "white",
    flexDirection: "column",
    flex: 1,
    paddingTop: 50
  },
  innerContainer: {
    alignItems: "center",
    backgroundColor: "white",
    justifyContent: "flex-start",
    paddingLeft: 20
  },
  image: {
    justifyContent: "center",
    alignItems: "center",
    marginTop: 60,
    width: null,
    height: 150,
    resizeMode: "contain",
    borderRadius: 8,
    flex: 1
  },
  text: {
    paddingTop: 5,
    fontSize: 14,
    fontWeight: "300",
    justifyContent: "center",
    alignItems: "center"
  },
  titleText: {
    fontSize: 36,
    fontWeight: "600",
    justifyContent: "center",
    alignItems: "center"
  },
  listingText: {
    fontSize: 24,
    fontWeight: "700",
    marginBottom: 20,
    marginTop: 20,
    paddingHorizontal: 20
  }
});
