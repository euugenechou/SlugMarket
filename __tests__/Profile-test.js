// __tests__/Profile-test.js
import 'react-native';
import React from 'react';
import Profile from '../src/screens/Profile'

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Profile />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});