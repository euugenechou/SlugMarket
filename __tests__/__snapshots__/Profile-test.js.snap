// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "backgroundColor": "white",
      }
    }
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    scrollEventThrottle={16}
    showsVerticalScrollIndicator={false}
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "flex": 1,
              "flexDirection": "row-reverse",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/Slug.png",
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 8,
                "flex": 1,
                "height": 150,
                "justifyContent": "center",
                "marginTop": 60,
                "resizeMode": "contain",
                "width": null,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "column",
                "paddingTop": 60,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "justifyContent": "flex-start",
                  "paddingLeft": 20,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignItems": "center",
                    "fontSize": 36,
                    "fontWeight": "600",
                    "justifyContent": "center",
                  }
                }
              >
                
                 
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignItems": "center",
                    "fontSize": 14,
                    "fontWeight": "300",
                    "justifyContent": "center",
                    "paddingTop": 5,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignItems": "center",
                    "fontSize": 14,
                    "fontWeight": "300",
                    "justifyContent": "center",
                    "paddingTop": 5,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 15,
                      "marginRight": 15,
                    },
                    Object {
                      "shadowColor": "rgba(0,0,0, .4)",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 1,
                      },
                      "shadowOpacity": 1,
                      "shadowRadius": 1,
                    },
                    Object {
                      "paddingTop": 10,
                      "width": 150,
                    },
                    Object {
                      "borderRadius": 5,
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={null}
                >
                  <View
                    pointerEvents="box-only"
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#9E9E9E",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "padding": 19,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "backgroundColor": "teal",
                        },
                        Object {
                          "borderRadius": 5,
                        },
                        Object {
                          "padding": 12,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 20,
                            },
                            Object {
                              "color": "white",
                            },
                            Object {
                              "fontSize": 17.5,
                            },
                            undefined,
                            undefined,
                            Object {
                              "fontWeight": "bold",
                            },
                            undefined,
                            undefined,
                          ],
                        ]
                      }
                    >
                      Sell Item
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "700",
              "marginBottom": 20,
              "marginTop": 20,
              "paddingHorizontal": 20,
            }
          }
        >
          My Listings
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "700",
              "marginBottom": 20,
              "marginTop": 20,
              "paddingHorizontal": 20,
            }
          }
        >
          My Sold Listings
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
